Jit/dialogs redesign

BeatC opened the pull request:
Rebased version of https://github.com/jitsi/jitsi-meet/pull/950

yanas commented on modules/UI/toolbars/Toolbar.js:
```diff
@@ -1,66 +1,12 @@
-        'Copy' // Focus Copy button.
-    );
-}
+let Toolbar;
```
Exposing the openLinkDialog function from Toolbar doesn't seem the right way to achieve this. The roomUrl, the updateRoomUrl and the openLinkDialog should be moved to a separate Invite.js and be independent of the Toolbar. While doing this let's also move InviteDialog together with this new Invite.js to a separate module "invite" instead of dialogs. Also it will be great if we rename roomUrl to inviteUrl.

BeatC replied:
Refactored this to singleton object Invite

BeatC commented on modules/UI/dialogs/InviteDialog.js:
```diff
@@ -0,0 +1,352 @@
+
+        this.password = options.password;
+        this.roomUrl = options.roomUrl || null;
+        this.emitter = options.emitter || null;
```
Changed to `APP.UI.addListener` and `APP.UI.emitEvent`. Does we use eventEmitter instance in UI object as a single event bus, right?

yanas replied:
Thanks and yes.

BeatC commented on modules/UI/dialogs/InviteDialog.js:
```diff
@@ -0,0 +1,352 @@
+    submitFunction(e, v) {
+        if (v && this.roomUrl) {
+            JitsiMeetJS.analytics.sendEvent('toolbar.invite.button');
+            this.emitter.emit(UIEvents.USER_INVITED, this.roomUrl);
```
Removed that event

BeatC commented on modules/UI/authentication/RoomLocker.js:
```diff
@@ -1,34 +1,37 @@
-/* global APP, JitsiMeetJS */
```
What do you mean? Move all RoomLocker module to Invite one or separate UI part from logic and move it to Invite module?

BeatC commented:
Removed room Locked/Unlocked button in the extended toolbar. Now this functionality is in Invite dialog. Finished refactoring of ContactList and Invite modules. Rebased against master and pushed. Because of removing element from toolbar there are 15 failed tests. But probably we should disable them as button doesn't exist in design now.

BeatC commented:
Tests are fixed in https://github.com/jitsi/jitsi-meet-torture/pull/49

yanas commented:
Merged manually.
